### PR TITLE
Log number of steps in execution (second attempt)

### LINKED ISSUE #1883: Log number of steps in execution
We should log the number of steps after execution or proving. The log entry should have `info` severity.

### PR DESCRIPTION
**Pull request by andreiburdusa**
---
Fixes #1883 
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Strategies/ProofState.hs [hunk 1]
```diff
@@ -59,20 +66,26 @@ instance Pretty Prim where
     pretty ApplyClaims = "apply claims"
     pretty ApplyAxioms = "apply axioms"
 
+newtype ExecutionDepth = ExecutionDepth { getDepth :: Natural}
```
Comment (ttuegel) [inline kore/src/Kore/Strategies/ProofState.hs:69]:
```suggestion
newtype ExecutionDepth = ExecutionDepth { getDepth :: Natural }
```

File: kore/src/Kore/Strategies/Verification.hs [hunk 2]
```diff
@@ -363,10 +365,24 @@ transitionRule' claims axioms =
...
+    :: TransitionRule m ReachabilityRule
+    -> TransitionRule m ReachabilityRule
+countExecutionDepth rule = \prim proofState -> do
+    result <- rule prim proofState
+    if prim `elem` [Prim.ApplyClaims, Prim.ApplyAxioms]
+        && isGoalRemainder result
+    then
+        fmap incrementDepth (rule prim proofState)
+    else
+        rule prim proofState
```
Comment (ttuegel) [inline kore/src/Kore/Strategies/Verification.hs:382]: This would call `rule` twice. It must return the `result` instead of calling `rule` again.

**Comment by andreiburdusa** [comment]
Turns out this is affecting the execution graph. `run-stepf-repl-script-spec.k` now generates this 
tooManyBranches
I still don't know where I changed the execution logic

**Comment by ttuegel** [comment]
Replaced by #1982.

**Pull request closed without merging**